### PR TITLE
Document options for meson

### PR DESCRIPTION
**Pull request by behlec**
Document what waring_level 1,2,3 means.

In response to #3275

### CODE REVIEW

**Comment by nirbheek** [comment]
This is all documented at: https://github.com/mesonbuild/meson/blob/master/docs/markdown/Builtin-options.md

But that page isn't listed in the sitemap so it's not showing up on `mesonbuild.com`. I was going to:

1. Add it to `docs/sitemap.txt`
2. Add a test to `run_unittests.py` next to `test_cpu_families_documented()` that ensures that all `docs/markdown/*.md` files are listed in `docs/sitemap.txt` so we don't miss this again

Do you want to do these and then edit the `Running-meson.md` page to link to `Builtin-options.md`?

**Comment by nirbheek** [comment]
Also, see: https://mesonbuild.com/Contributing.html#documentation you need to skip the CI for docs-only changes. That's why it's failing.

**Comment by behlec** [comment]
I will make the changes.

File: run_unittests.py [hunk 1]
```diff
@@ -2271,6 +2271,21 @@ def test_cpu_families_documented(self):
...
 
+    @unittest.skipIf(not os.path.isdir('docs'), 'Doc dir not found, presumably because this is a tarball release.')
+    def test_markdown_files_in_sitemap(self):
+        '''
+        Test that each markdown files in docs/markdown is referenced in sitemap.txt
+        '''
+        with open("docs/sitemap.txt") as f:
+            md = f.read()
+        self.assertIsNotNone(md)
+        toc = list(m.group(1) for m in re.finditer(r"^\s*(\w.*)$",md, re.MULTILINE))
```
Comment (ignatenkobrain) [inline run_unittests.py]: __[Flake8]__

[E231] missing whitespace after ','

File: docs/markdown/Builtin-options.md [hunk 2]
```diff
@@ -25,21 +25,31 @@ Installation options are all relative to the prefix, except:
...
+| sharedstatedir                       | com           | Architecture-independent data directory |
+| werror                               | false         | Treat warnings as erros |
+| warnlevel {1, 2, 3}                  | 1             | Set the warning level. From 1 = lowest to 3 = highest |
+| layout {mirror,flat}                 | mirror        | Build directory layout. |
+| default-library {shared, static, both} | shared       | Default library type. |
+| backend {ninja, vs,<br>vs2010, vs2015, vs2017, xcode} |               | Backend to use (default: ninja). |
+| stdsplit                             |               | Split stdout and stderr in test logs. |
+| errorlogs                            |               | Whether to print the logs from failing tests. |
+| cross-file CROSS_FILE                |               | File describing cross compilation environment. |
+| wrap-mode {WrapMode.default,<br>WrapMode.nofallback, WrapMode.nodownload,<br>WrapMode.forcefallback} | | Special wrap mode to use |
```
Comment (nirbheek) [inline docs/markdown/Builtin-options.md]: This should be `{default, nofallback, nodownload, forcefallback}`.

Also the table does not need to end with `|` afaik.

File: docs/markdown/Builtin-options.md [hunk 3]
```diff
@@ -62,19 +72,19 @@ a builddir and then run `meson configure` on it with no options.
...
+| b_bitcode   | false         | true, false             | Embed Apple bitcode, see below |
+| b_colorout  | always        | auto, always, never     | Use colored output |
+| b_coverage  | false         | true, false             | Enable coverage tracking |
+| b_lundef    | true          | true, false             | Don't allow undefined symbols when linking |
+| b_lto       | false         | true, false             | Use link time optimization |
+| b_ndebug    | false         | true, false, if-release | Disable asserts |
+| b_pch       | true          | true, false             | Use precompiled headers |
+| b_pgo       | off           | off, generate, use      | Use profile guided optimization |
+| b_sanitize  | none          | see below               | Code sanitizer to use |
+| b_staticpic | true          | true, false             | Build static libraries as position independent |
```
Comment (nirbheek) [inline docs/markdown/Builtin-options.md]: Table does not need to end with `|`, afaik.

File: docs/markdown/Builtin-options.md [hunk 4]
```diff
@@ -97,18 +107,18 @@ The following options are available. Note that both the options themselves and
...
+| c_args       |               | free-form comma-separated list           | C compile arguments to use |
+| c_link_args  |               | free-form comma-separated list           | C link arguments to use |
+| c_std        | none          | none, c89, c99, c11, gnu89, gnu99, gnu11 | C language standard to use |
+| c_winlibs    | see below     | free-form comma-separated list           | Standard Windows libs to link against |
+| cpp_args     |               | free-form comma-separated list           | C++ compile arguments to use |
+| cpp_link_args|               | free-form comma-separated list           | C++ link arguments to use |
+| cpp_std      | none          | none, c++98, c++03, c++11, c++14, c++17, <br/>c++1z, gnu++03, gnu++11, gnu++14, gnu++17, gnu++1z | C++ language standard to use |
+| cpp_debugstl | false         | true, false                              | C++ STL debug mode |
+| cpp_eh       | sc            | none, a, s, sc                           | C++ exception handling type |
+| cpp_winlibs  | see below     | free-form comma-separated list           | Standard Windows libs to link against |
```
Comment (nirbheek) [inline docs/markdown/Builtin-options.md]: Same as above.

**Comment by behlec** [comment]
I made the changes. There is a file `Module-reference.md` which is not contained. I think this file is not used anymore, modules are documented in other files.

File: docs/sitemap.txt [hunk 5]
```diff
@@ -5,6 +5,7 @@ index.md
 	Manual.md
 		Overview.md
 		Running-Meson.md
+			Builtin-options.md
```
Comment (nirbheek) [inline docs/sitemap.txt]: This should be at the same level as `Running-Meson.md`, not underneath it. That just makes it harder to find.

**Comment by nirbheek** [comment]
> I made the changes. There is a file Module-reference.md which is not contained. I think this file is not used anymore, modules are documented in other files.

I guess this should be removed, @jpakkane?

Also, you added [skip ci] to all the commits, even the one that actually added a test, which doesn't make sense. Please squash all the commits into one commit and remove `[skip ci]` from it.

File: docs/markdown/Builtin-options.md (hunk 2 above)
Comment (behlec) [reply docs/markdown/Builtin-options.md]: The table must end with |

Comment (nirbheek) [reply docs/markdown/Builtin-options.md]: Okay :)

**Pull request merged**